tests/components/prometheus/test_init.py

```diff
@@ -1665,14 +1662,25 @@ async def test_disabling_entity(
     ).withValue(1).assert_in_metrics(body)
 
 
-@pytest.mark.parametrize("namespace", [""])
-async def test_entity_becomes_unavailable_with_export(
+@pytest.mark.parametrize("unavailable_state", [STATE_UNAVAILABLE, STATE_UNKNOWN])
+async def test_entity_becomes_unavailable(
     hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
     entity_registry: er.EntityRegistry,
-    client: ClientSessionGenerator,
     sensor_entities: dict[str, er.RegistryEntry],
+    unavailable_state: str,
 ) -> None:
-    """Test an entity that becomes unavailable is still exported."""
+    """Test an entity that becomes unavailable/unknown is no longer exported."""
+    reset_prometheus_registry()
+    config = {
+        prometheus.CONF_PROM_NAMESPACE: "",
+    }
+    assert await async_setup_component(
+        hass, prometheus.DOMAIN, {prometheus.DOMAIN: config}
+    )
+    await hass.async_block_till_done()
+    client = await hass_client()
```

Comment:
I don't quite understand the reasons for resetting the registry and re-initializing the component here. I tried to [remove it](https://knyar.net/paste/384ded69), and the tests were still passing. Perhaps this was necessary when new behavior was behind a configuration flag, but it's not anymore?

Reply:
Yes, it was some kind of leftover code. I've removed it. Thanks!